src/pycell.rs

```diff
@@ -392,11 +395,11 @@ impl<T: PyClass> PyCell<T> {
     /// });
     /// ```
     pub fn try_borrow(&self) -> Result<PyRef<'_, T>, PyBorrowError> {
-        let flag = self.get_borrow_flag();
+        let flag = crate::class::impl_::BorrowImpl::get_borrow_flag()(self);
         if flag == BorrowFlag::HAS_MUTABLE_BORROW {
             Err(PyBorrowError { _private: () })
         } else {
-            self.set_borrow_flag(flag.increment());
+            crate::class::impl_::BorrowImpl::increment_borrow_flag()(self, flag);
             Ok(PyRef { inner: self })
         }
```

Comment:
With this design, the implementation of this then becomes:

```rust
self.borrow_checker().borrow().map(|_| PyRef { inner: self })
```

and similar for the other methods.